### PR TITLE
Vishwajit/typescript support changes

### PR DESCRIPTION
**Pull request by CharmisG**
I had upgrade plugin as well as version for android and iOS and also convert code into typescript.
Please review my PR.

### CODE REVIEW

File: react-native-in-app-payments-quickstart/README.md [hunk 1]
```diff
@@ -111,3 +111,6 @@ Square provides a sample backend service for this quickstart that can easily be
 [root README]: ../README.md
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/README.md:111]: After the upgrade, do we need update the steps in this README?

File: react-native-in-app-payments-quickstart/android/app/src/main/java/sqip/react/example/MainApplication.java [hunk 2]
```diff
@@ -43,5 +44,37 @@ public ReactNativeHost getReactNativeHost() {
   public void onCreate() {
     super.onCreate();
     SoLoader.init(this, /* native exopackage */ false);
+    // initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/android/app/src/main/java/sqip/react/example/MainApplication.java:47]: Do we suppose to comment this line?

File: react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml [hunk 3]
```diff
@@ -8,7 +8,7 @@
     </style>
 
     <style name="sqip.Theme.CardEntry" parent="AppTheme">
-        <item name="sqipSaveButtonText">@string/pay_button</item>
+<!--        <item name="sqipSaveButtonText">@string/pay_button</item>-->
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml:11]: Why do we comment this line?

File: react-native-in-app-payments-quickstart/app/components/AddressView.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ AddressView.propTypes = {
   address: PropTypes.string.isRequired,
 };
 
-export default function AddressView({ address }) {
+export default function AddressView({ address} : { address: any}) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/AddressView.tsx:27]: should address be string?

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx [hunk 5]
```diff
@@ -29,21 +29,21 @@ CardsOnFileCardView.propTypes = {
   onSelectCardOnFile: PropTypes.func.isRequired,
 };
 
-function showConfirmation(cardOnFile, onConfirm) {
+function showConfirmation(cardOnFile: any, onConfirm: any) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx:32]: can we use the explicit type when it's possible?

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx [hunk 6]
```diff
@@ -29,21 +29,21 @@ CardsOnFileCardView.propTypes = {
...
+    },
+    {
+      text: 'Purchase',
+      onPress: onConfirm,
+    },
+  ]);
 }
 
-export default function CardsOnFileCardView({ cardOnFile, onSelectCardOnFile }) {
+export default function CardsOnFileCardView({ cardOnFile }: { cardOnFile: any }, { onSelectCardOnFile }: { onSelectCardOnFile: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx:46]: please use a concrete type instead of `any`

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx [hunk 7]
```diff
@@ -19,25 +19,24 @@ import {
...
 import CardsOnFileCardView from './CardsOnFileCardView';
 import GreenButton from './GreenButton';
 
-CardsOnFileModal.propTypes = {
-  cardsOnFile: PropTypes.array.isRequired,
-  onSelectCardOnFile: PropTypes.func.isRequired,
-  onCloseCardsOnFileScreen: PropTypes.func.isRequired,
-  onShowCustomerCardEntry: PropTypes.func.isRequired,
+interface Props {
+  onCloseCardsOnFileScreen: any,
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx:28]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx [hunk 8]
```diff
@@ -46,21 +45,20 @@ export default function CardsOnFileModal({
           ? (
             <Text style={styles.noCardsText}>
               {'No cards on file found.'
-              + 'Please tap the button to add a card that you can use for future transactions.'}
+                + 'Please tap the button to add a card that you can use for future transactions.'}
             </Text>
-          ) : cardsOnFile.map(cardOnFile => (
+          ) : cardsOnFile.map((cardOnFile: any) => (
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx:50]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileTitleView.tsx [hunk 9]
```diff
@@ -30,7 +30,7 @@ CardsOnFileTitleView.propTypes = {
   onCloseCardsOnFileScreen: PropTypes.func.isRequired,
 };
 
-export default function CardsOnFileTitleView({ onCloseCardsOnFileScreen }) {
+export default function CardsOnFileTitleView({ onCloseCardsOnFileScreen }: { onCloseCardsOnFileScreen: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CardsOnFileTitleView.tsx:33]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/DigitalWalletButton.tsx [hunk 10]
```diff
@@ -29,7 +29,7 @@ DigitalWalletButton.propTypes = {
   onPress: PropTypes.func.isRequired,
 };
 
-export default function DigitalWalletButton({ onPress }) {
+export default function DigitalWalletButton({ onPress }: { onPress: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/DigitalWalletButton.tsx:32]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/GreenButton.tsx [hunk 11]
```diff
@@ -21,12 +21,18 @@ import {
 } from 'react-native';
 import PropTypes from 'prop-types';
 
-GreenButton.propTypes = {
-  onPress: PropTypes.func.isRequired,
-  text: PropTypes.string.isRequired,
+interface GreenButton {
+  onPress: any,
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/GreenButton.tsx:25]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/GreenButton.tsx [hunk 12]
```diff
@@ -21,12 +21,18 @@ import {
...
+  text: any,
 };
 
-export default function GreenButton({ onPress, text }) {
+// GreenButton.propTypes = {
+//   onPress: PropTypes.func.isRequired,
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/GreenButton.tsx:30]: delete?

File: react-native-in-app-payments-quickstart/app/components/OrderInformationTitleView.tsx [hunk 13]
```diff
@@ -24,7 +24,7 @@ OrderInformationTitleView.propTypes = {
   title: PropTypes.string.isRequired,
 };
 
-export default function OrderInformationTitleView({ title }) {
+export default function OrderInformationTitleView({ title }: { title: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/OrderInformationTitleView.tsx:27]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/OrderInformationView.tsx [hunk 14]
```diff
@@ -26,7 +26,7 @@ OrderInformationView.propTypes = {
   description: PropTypes.string.isRequired,
 };
 
-export default function OrderInformationView({ title, description }) {
+export default function OrderInformationView({ title }: { title: any }, { description }: { description: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/OrderInformationView.tsx:29]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/OrderModal.tsx [hunk 15]
```diff
@@ -28,23 +28,23 @@ import AddressView from './AddressView';
...
 import DigitalWalletButton from './DigitalWalletButton';
 
-OrderModal.propTypes = {
-  onCloseOrderScreen: PropTypes.func.isRequired,
-  onPayWithGiftCard: PropTypes.func.isRequired,
-  onPayWithCard: PropTypes.func.isRequired,
-  onShowDigitalWallet: PropTypes.func.isRequired,
-  onBuyerVerification: PropTypes.func.isRequired,
+interface OrderModal {
+  onCloseOrderScreen: any,
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/OrderModal.tsx:32]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/components/OrderTitleView.tsx [hunk 16]
```diff
@@ -30,7 +30,7 @@ OrderTitleView.propTypes = {
   onCloseOrderScreen: PropTypes.func.isRequired,
 };
 
-export default function OrderTitleView({ onCloseOrderScreen }) {
+export default function OrderTitleView({ onCloseOrderScreen }: { onCloseOrderScreen: any }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/OrderTitleView.tsx:33]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 17]
```diff
@@ -147,57 +120,56 @@ export default class HomeScreen extends Component {
...
-    this.setState({
-      canUseDigitalWallet: digitalWalletEnabled,
-    });
+    setcanUseDigitalWallet(digitalWalletEnabled)
   }
 
-  async onApplePayRequestNonceSuccess(cardDetails) {
-    if (this.chargeServerHostIsSet()) {
+
+  const onApplePayRequestNonceSuccess = async (cardDetails: any) => {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:127]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 18]
```diff
@@ -209,23 +181,23 @@ export default class HomeScreen extends Component {
     }
   }
 
-  onGooglePayRequestNonceFailure(errorInfo) {
+  const onGooglePayRequestNonceFailure = (errorInfo: any) => {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:184]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 19]
```diff
@@ -147,57 +120,56 @@ export default class HomeScreen extends Component {
...
+      showAlert('Error processing Apple Pay payment', applePayError);
     } else { // the state is none, so they canceled
-      this.showOrderScreen();
+      showOrderScreen();
     }
   }
 
-  async onGooglePayRequestNonceSuccess(cardDetails) {
-    if (this.chargeServerHostIsSet()) {
+  const onGooglePayRequestNonceSuccess = async (cardDetails: any) => {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:166]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/service/Charge.tsx [hunk 20]
```diff
@@ -16,7 +16,7 @@
 import { CHARGE_SERVER_URL } from '../Constants';
 import ChargeError from '../ChargeError';
 
-export default async function chargeCardNonce(nonce, verificationToken) {
+export default async function chargeCardNonce(nonce: any, verificationToken = undefined) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/service/Charge.tsx:19]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.tsx [hunk 21]
```diff
@@ -17,7 +17,7 @@
 import { CHARGE_CUSTOMER_CARD_SERVER_URL } from '../Constants';
 import ChargeCustomerCardError from '../ChargeCustomerCardError';
 
-export default async function chargeCustomerCard(customer_id, customer_card_id) {
+export default async function chargeCustomerCard(customer_id: any, customer_card_id: any) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.tsx:20]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/app/service/CreateCustomerCard.tsx [hunk 22]
```diff
@@ -17,7 +17,7 @@
 import { CREATE_CUSTOMER_CARD_SERVER_URL } from '../Constants';
 import CreateCustomerCardError from '../CreateCustomerCardError';
 
-export default async function createCustomerCard(customer_id, nonce) {
+export default async function createCustomerCard(customer_id: any, nonce: any) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/service/CreateCustomerCard.tsx:20]: please use a concrete type instead of any

File: react-native-in-app-payments-quickstart/index.js [hunk 23]
```diff
@@ -1,20 +1,9 @@
-/*
- Copyright 2019 Square Inc.
+/**
+ * @format
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/index.js:2]: nit: please add the header.

File: react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart/LaunchScreen.storyboard [hunk 24]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart/LaunchScreen.storyboard:1]: Is this file required?

File: src/models/Card.ts [hunk 25]
```diff
@@ -0,0 +1,20 @@
+import Brand from './Brand';
+import PrepaidType from './PrepaidType';
+import Type from './Type';
+
+interface Card {
+  brand?: Brand;
```
Comment (hukid) [inline src/models/Card.ts:6]: Are these fields optional?

File: src/models/CardDetails.ts [hunk 26]
```diff
@@ -0,0 +1,8 @@
+import Card from './Card';
+
+interface CardDetails {
+  nonce?:string;
```
Comment (hukid) [inline src/models/CardDetails.ts:4]: Are these fields optional?

File: src/models/VerificationResult.ts [hunk 27]
```diff
@@ -0,0 +1,7 @@
+interface VerificationResult {
+  nonce?: string;
```
Comment (hukid) [inline src/models/VerificationResult.ts:2]: are these fields optional?

**Review by hukid — CHANGES_REQUESTED** [review]

File: react-native-in-app-payments-quickstart/README.md (hunk 1 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/README.md:111]: No it's not require.

File: react-native-in-app-payments-quickstart/android/app/src/main/java/sqip/react/example/MainApplication.java (hunk 2 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/android/app/src/main/java/sqip/react/example/MainApplication.java:47]: I had removed this line

File: react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml (hunk 3 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml:11]: Because after plugin upgration it's not supported.

File: react-native-in-app-payments-quickstart/app/components/AddressView.tsx (hunk 4 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/AddressView.tsx:27]: Changed to String.

File: src/models/VerificationResult.ts (hunk 27 above)
Comment (CharmisG) [reply src/models/VerificationResult.ts:2]: No it's not.

File: src/models/CardDetails.ts (hunk 26 above)
Comment (CharmisG) [reply src/models/CardDetails.ts:4]: No it's not.

File: src/models/Card.ts (hunk 25 above)
Comment (CharmisG) [reply src/models/Card.ts:6]: No it will use in CardDetail screen.

File: react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart/LaunchScreen.storyboard (hunk 24 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart/LaunchScreen.storyboard:1]: Yes It is.

File: react-native-in-app-payments-quickstart/index.js (hunk 23 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/index.js:2]: Done

File: react-native-in-app-payments-quickstart/app/components/GreenButton.tsx (hunk 12 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/GreenButton.tsx:30]: Yes Done.

File: react-native-in-app-payments-quickstart/app/service/CreateCustomerCard.tsx (hunk 22 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/service/CreateCustomerCard.tsx:20]: Resolved it.

File: react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.tsx (hunk 21 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.tsx:20]: Resolved it

File: react-native-in-app-payments-quickstart/app/service/Charge.tsx (hunk 20 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/service/Charge.tsx:19]: Resolved it.

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 19 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:166]: Resolved it.

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 17 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:127]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx (hunk 5 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx:32]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx (hunk 6 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CardsOnFileCardView.tsx:46]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx (hunk 7 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx:28]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileTitleView.tsx (hunk 9 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CardsOnFileTitleView.tsx:33]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/DigitalWalletButton.tsx (hunk 10 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/DigitalWalletButton.tsx:32]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/OrderInformationView.tsx (hunk 14 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/OrderInformationView.tsx:29]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/OrderInformationTitleView.tsx (hunk 13 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/OrderInformationTitleView.tsx:27]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/OrderModal.tsx (hunk 15 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/OrderModal.tsx:32]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/OrderTitleView.tsx (hunk 16 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/OrderTitleView.tsx:33]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/GreenButton.tsx (hunk 11 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/GreenButton.tsx:25]: Resolved it.

File: react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx (hunk 8 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CardsOnFileModal.tsx:50]: Resolved it.

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 18 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:184]: Resolved it.

File: react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml (hunk 3 above)
Comment (hukid) [reply react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml:11]: Shall we just remove it?

File: src/models/CardDetails.ts (hunk 26 above)
Comment (hukid) [reply src/models/CardDetails.ts:4]: If they are not optional, should we make these fields required?

File: react-native-in-app-payments-quickstart/app/Utilities.tsx [hunk 28]
```diff
@@ -62,9 +62,9 @@ export function printCurlCommand(nonce: any, appId: any, verificationToken = und
   }
 }
 
-export async function showAlert(title: any, message = "", onPress = () => { }) {
+export async function showAlert(title: string, message = "", onPress = () => { }) {
   Alert.alert(
-    title,
+    title.toString(),
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/Utilities.tsx:67]: if title is a string type, why do we still need a `.toString()`?

**Review by hukid — APPROVED** [review]
a few small comments, otherwise looks good! Thanks!

File: react-native-in-app-payments-quickstart/app/Utilities.tsx (hunk 28 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/Utilities.tsx:67]: Removed.

File: src/models/CardDetails.ts (hunk 26 above)
Comment (CharmisG) [reply src/models/CardDetails.ts:4]: We had already add ? sign for require fields.

File: react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml (hunk 3 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/android/app/src/main/res/values/themes.xml:11]: Removed it.

File: src/models/VerificationResult.ts (hunk 27 above)
Comment (hukid) [reply src/models/VerificationResult.ts:2]: I mean the fields, not the object. Is `nonce` or `token` optional or not?

Comment (CharmisG) [reply src/models/VerificationResult.ts:2]: no it's not
